pkgs/by-name/s3/s3proxy/package.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, fetchFromGitHub
+, jre
+, makeWrapper
+, maven
+
+, version ? "2.1.0"
+, hash ? "sha256-GhZPvo8wlXInHwg8rSmpwMMkZVw5SMpnZyKqFUYLbrE="
+, mvnHash ? "sha256-85mE/pZ0DXkzOKvTAqBXGatAt8gc4VPRCxmEyIlyVGI="
+}:
+
+let
+  pname = "s3proxy";
+in
+maven.buildMavenPackage {
+  inherit pname version mvnHash;
+
+  src = fetchFromGitHub {
+    inherit hash;
```

Comment:
```suggestion
}:
maven.buildMavenPackage {
  pname = "s3proxy";
  version = "2.1.0";
  mvnHash = "sha256-85mE/pZ0DXkzOKvTAqBXGatAt8gc4VPRCxmEyIlyVGI=";

  src = fetchFromGitHub {
    hash = "sha256-GhZPvo8wlXInHwg8rSmpwMMkZVw5SMpnZyKqFUYLbrE=";
```

Reply:
I don't think defining `version`, `mvnHash` and  `hash` as parameter is a good way to define your package. The standard way I can think is merge everything inside the `buildMavenPackage` scope and set the `hash` inside the `fetchFromGithub` :+1: .